src/mon/PGMonitor.cc

```diff
@@ -190,6 +190,8 @@ void PGMonitor::update_from_paxos(bool *need_bootstrap)
     int r = get_version(pg_map.version + 1, bl);
     if (r == -ENOENT) {
       dout(10) << __func__ << " failed to read_incremental, read_full" << dendl;
+      // reset pg map
+      pg_map = PGMap();
```

Comment:
I think this will work as the patch intends.

Looking at this function however, made me realize we are not taking any steps if the incremental (or the full version) do not exist. @tchaikov do you know if that's by design or just something that (at somepoint) was overlooked?